app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -31,6 +37,7 @@ class DetailedItemFragment : Fragment() {
         super.onCreate(savedInstanceState)
         observer = ImageAccess(requireActivity())
         lifecycle.addObserver(observer)
+        userViewModel.setUser(currentUserProvider.getCurrentUserId())
```

Comment:
Why do you set the user of the viewmodel here? We also set it elsewhere to correspond to the user that owns the item. Also since in your methods wishlistContains, modifyWishlist and refreshUi you call the currentUserProvider, you don't need to do it here.